toolchain/sem_ir/inst.h

```diff
@@ -88,31 +136,40 @@ class Inst : public Printable<Inst> {
   }
 
   // Returns whether this instruction has the specified type.
-  template <typename TypedInst>
+  template <typename TypedInst, typename Info = InstLikeTypeInfo<TypedInst>>
   auto Is() const -> bool {
-    return kind() == TypedInst::Kind;
+    return Info::IsKind(kind());
   }
 
   // Casts this instruction to the given typed instruction, which must match the
   // instruction's kind, and returns the typed instruction.
-  template <typename TypedInst, typename Info = TypedInstArgsInfo<TypedInst>>
+  template <typename TypedInst, typename Info = InstLikeTypeInfo<TypedInst>>
   auto As() const -> TypedInst {
-    CARBON_CHECK(Is<TypedInst>()) << "Casting inst of kind " << kind()
-                                  << " to wrong kind " << TypedInst::Kind;
-    auto build_with_type_id_and_args = [&](auto... type_id_and_args) {
+    CARBON_CHECK(Is<TypedInst>())
+        << "Casting inst of kind " << kind() << " to wrong kind "
+        << typeid(TypedInst).name();
```

Comment:
My experience was that typeid names were not pretty. Can we get the kind name in those cases where it is a single kind by having a name member of the info type?

Reply:
Done.